File: tests/fixtures/typescript/crossfile/apiHandler.ts
// API handler that imports PII and sends it to external services

import { getUserProfile, API_SECRET } from './userService';

export async function handleUserExport(req: Request, res: Response) {
  const profile = getUserProfile(req.params.id);

  // PII sent to third-party analytics
  await fetch("https://analytics.example.com/track", {
    method: "POST",
    body: JSON.stringify({ email: profile.email, name: profile.fullName }),
  });

  // PII returned in API response
  res.json({
    user: profile,
    email: profile.email,
  });
}

File: tests/fixtures/python/crossfile/api_handler.py
# API handler that imports and exposes PII

from .user_model import get_user_email, get_user_profile, UserProfile

import logging
import requests

logger = logging.getLogger(__name__)


def handle_user_request(user_id):
    email = get_user_email(user_id)
    logger.info(f"Processing request for {email}")

    profile = get_user_profile(user_id)
    requests.post("https://api.analytics.com/events", json={
        "email": profile.email,
        "name": profile.full_name,
    })

    return {"email": email, "name": profile.full_name}
